File: Python/1446.py
class Solution:
    def maxPower(self, s: str) -> int:
        # String length of 0, 1
        s_len = len(s)
        if s_len == 0: return 0
        if s_len == 1: return 1
        
        tracker = s[0]
        farMax = 0; count = 0; i = 0
        while s_len > i:
            if s[i] == tracker: 
                count += 1
            else:
                # New character
                tracker = s[i]
                farMax = max(farMax, count)
                count = 1
            i += 1
        # Incase s = one characters
        farMax = max(farMax, count)
                       
        return farMax
        
        

File: CPP/1446.cpp
class Solution {
public:
    int maxPower(string s) {
        // Null string,  length 1
        if (s.size() == 0) return 0;
        else if (s.size() == 1) return 1;
        
        char tracker = s[0];
        int farMax = 0, count = 0;
        for (char c : s){
            if (c == tracker) count++;
            else{
                 // Update with new character
                tracker = c;
                // Update max value
                farMax = max(farMax, count);
                count = 1;
            }
        }
        // Incase the string consists of one characters.
        farMax = max(farMax, count);
        return farMax;
    }
};